sdv/model.py

```diff
@@ -841,11 +667,159 @@ async def get(self) -> TypedDataPointResult[List[str]]:
             raise
 
     async def set(self, value: List[str]):
+        await self._set(value, self.__class__.__name__)
+
+    def create_broker_data_point(self, value: List[str]):
+        array = StringArray(values=value)
+        return BrokerDatapoint(string_array=array)
+
+
+class BatchSetBuilder:
+    """Collects data points to be updated in a single 'atomic' set"""
+
+    def __init__(self, client):
+        self.__client = client
+        self.__nodes = {}
+
+    def preset(self, node: DataPoint, value) -> "BatchSetBuilder":
+        node_name = node.get_path()
+        node_value = node.create_broker_data_point(value)
+        if node_name in self.__nodes:
+            logger.error(
+                "Key '%s' already present in set-batch!"
+                " Overwriting existing value '%s' with '%s'.",
+                node_name,
+                self.__nodes[node_name],
+                node_value,
+            )
+        self.__nodes[node_name] = node_value
+        return self
+
+    async def apply(self):
+        if len(self.__nodes) == 0:
+            logger.warning("Empty node list, nothing updated")
+            return
+
         try:
-            array = StringArray(values=value)
-            datapoint = BrokerDatapoint(string_array=array)
-            await self._set(datapoint)
-        except (grpc.aio.AioRpcError, Exception) as ex:  # type: ignore
-            logger.error("Error occured in DataPointStringArray.set")
-            logger.exception(ex)
+            response = await self.__client.SetDatapoints(self.__nodes)
+            if response.errors:
+                raise TypeError(
+                    "Some data point values could not be set: ", response.errors
+                )
+
+        except (grpc.aio.AioRpcError, Exception):  # type: ignore
+            logger.error("Error occured on updating several data points")
             raise
+
+
+class Model(Node):
+    """The Model class represents a branch of the model tree, including root.
+    Leafs are typcially one of the typed DataPoint* classes.
+    But also a Model class can be a leaf, if it does not contain data Points,
+    just methods."""
+
+    def preset(self, node: DataPoint, value) -> BatchSetBuilder:
```

Comment:
with this, we would be able to call preset on any leaf in the vehicle model tree. But we only want to be able to do it on the root / vehicle

Reply:
Problem is, that we don't have a defined model root in the SDK, only in the generated concrete model.
I personally don't understand why we need a base class called model in parallel to the Node class. In my opinion it would be fine to use the `Node` class as any intermediate node/branch of the model and have `Model`as the defined root of the whole tree ...
For the moment I would leave it that way till we do the overall refactoring of the model.

Reply:
OK for me